plugins/wasm-cpp/extensions/jwt_auth/README_EN.md

```diff
@@ -0,0 +1,399 @@
+
+4. The gateway returns the response with the token to the client, and the client needs to cache this token locally;
+
+5. The client sends a business request to the API gateway, carrying the token in the request;
+
+6. The gateway uses the public key set by the user to verify the token in the request. After the verification is passed, the request is passed through to the backend service;
+
+7. The backend service processes the business and responds;
+
+8. The gateway returns the business response to the client.
+
+In this entire process, the gateway uses the token authentication mechanism to realize the ability of users to use their own user system to authorize their API. Next, we will introduce the structured token Json Web Token (JWT) used by the gateway to implement token authentication.
+
+### 1.3 JWT
+
+#### 1.3.1 Introduction
+
+Json Web Token (JWT) is an open standard RFC7519 based on JSON for executing a type of claim to pass between network applications. JWT can generally be used as an independent identity verification token, which can contain user identification, user roles, and permissions information, making it easier to obtain resources from the resource server, and can also add some other necessary declarative information for other business logic, especially suitable for the login scenario of distributed sites.
+
+#### 1.3.2 Composition of JWT
+
+`eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiYWRtaW4iOnRydWV9.TJVA95OrM7E2cBab30RMHrHDcEfxjoYZgeFONFh7HgQ`
+
+As shown in the example above, JWT is a string consisting of three parts:
+
+- Header
+- Payload
+- Signature
+
+**Header**
+
+The header of the JWT carries two pieces of information:
+
+- The type of the token, which is JWT
+- The algorithm used for encryption
+
+The gateway supports the following encryption algorithms:
+
+```text
+ES256, ES384, ES512,
+HS256, HS384, HS512,
+RS256, RS384, RS512,
+PS256, PS384, PS512,
+EdDSA
+```
+
+The complete header looks like the following JSON:
+
+```js
+{
+  'typ': 'JWT',
+  'alg': 'HS256'
+}
+```
+
+The header is then Base64-encoded (this encoding can be symmetrically decoded), forming the first part:
+
+`eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9`
+
+**Payload**
+
+The payload is where the actual information is stored. The details are defined below:
+
+```text
+iss: The issuer of the token. This indicates who created the token and is a string value.
+sub: The subject identifier. This is the unique identifier for the end user provided by the issuer, and is no longer than 255 ASCII characters, and is case-sensitive within the issuer's scope.
+aud: The audience(s) of the token, which is an array of case-sensitive strings.
+exp: The expiration time of the token, which is a Unix timestamp in seconds, and any token that is issued after that time is invalid.
```

Comment:
`any token that is issued after that time is invalid`, I don't think it is correct.